components/scream/src/share/scream_logging.hpp

```diff
@@ -0,0 +1,271 @@
+
+#ifndef SCREAM_LOGGING_HPP
+#define SCREAM_LOGGING_HPP
+
+#include <cstdio>
+#include <cstdlib>
+#include <string>
+#include <vector>
+#include <stdexcept>
+#include "scream_config.h"
+
+#if LOG_STACK_TRACE
+#include <execinfo.h>
+#endif
+
+namespace scream {
+namespace util {
+struct Error : public std::runtime_error {
+  explicit Error(const std::string &s) : std::runtime_error(s) {}
+};
+}  // namespace util
+}  // namespace scream
+
+#include <assert.h>
+#include <iostream>
```

Comment:
Can we keep all includes at the top, and avoid breaking namespaces just to re-open them right after? The way it is makes reading the file harder, imo.